specification.markdown

```diff
@@ -1,66 +1,140 @@
-### EBML specifications
+# EBML specifications
 
-#### EBML principle
+## Abstract
 
-EBML is short for Extensible Binary Meta Language. EBML specifies a
-binary and octet (byte) aligned format inspired by the principle of XML.
-EBML itself is a generalized description of the technique of binary
-markup. Like XML, it is completely agnostic to any data that it might
-contain. Therein, the Matroska project is a specific implementation
-using the rules of EBML: It seeks to define a subset of the EBML
-language in the context of audio and video data (though it obviously
-isn't limited to this purpose). The format is made of 2 parts: the
-semantic and the syntax. The semantic specifies a number of IDs and
-their basic type and is not included in the data file/stream.
+This document defines the Extensible Binary Markup Language (EBML) file
+format. EBML is designed to be a binary format, inspired by XML, for the
+purpose of storing and the possibility to manipulate all sorts of
+digital content and metadata in a compact and simple form by using a
+hierarchical form with variable field length.
 
-Just like XML, the specific "tags" (IDs in EBML parlance) used in an
-EBML implementation are arbitrary. However, the semantic of EBML
-outlines general data types and ID's.
+Some of the advantages of EBML are:
+
+- Possibility for compatibility between different versions of binary
+    languages defined in EBML. A rare property of binary format that
+    otherwise often needs careful consideration beforehand.
+
+- Unlimited size of data payload.
+
+- Can be both generated and read as a stream, without knowing the data
+    size beforehand.
+
+- Often very space efficient, even compared to other binary
+    representations of the same data.
+
+Some of the EBML disadvantages are:
+
+- No references can be made between EBML files, such as includes or
+    inherits. Every EBML document is a self contained entity. The data
+    stored in EBML may of course reference other resources.
+
+- No compositioning process to merge two or more EBML languages
+    currently exists.
+
+This document describes the EBML binary syntax, its semantic
+interpretation and the syntax of a textual document type definition
+format used to define the rules and constraints of an EBML language.
+
+## Notations and Conventions
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this
+document are to be interpreted as described in RFC 2119.
```

Comment:
fixed in https://github.com/MediaArea/ebml-specification/commit/db1042fb059c14a34578c025462bc90c973e7a95